src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -140,23 +164,36 @@ protected override void ProcessRecord()
 
             try
             {
-                var parsedJson = JToken.Parse(Json);
+                var parsedJson = JsonNode.Parse(Json);
 
                 if (_jschema != null)
                 {
-                    var errorMessages = _jschema.Validate(parsedJson);
-                    if (errorMessages != null && errorMessages.Count != 0)
+                    var validationResults = _jschema.Validate(parsedJson, new ValidationOptions { OutputFormat = OutputFormat.Basic });
+                    result = validationResults.IsValid;
+                    if (!result)
                     {
-                        result = false;
-
                         Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
 
-                        foreach (var message in errorMessages)
+                        if (validationResults.Message != null)
                         {
-                            ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
-                            errorRecord.ErrorDetails = new ErrorDetails(message.ToString());
+                            ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema",
+                                ErrorCategory.InvalidData, null);
+                            var message = $"{validationResults.Message} at {validationResults.InstanceLocation}";
+                            errorRecord.ErrorDetails = new ErrorDetails(message);
                             WriteError(errorRecord);
                         }
+
+                        if (validationResults.HasNestedResults)
+                        {
+                            foreach (var nestedResult in validationResults.NestedResults.Where(x => x.Message != null))
```

Comment:
We prefer to avoid LINQ. Please replace with `if` operator in the loop body.

Reply:
Done.

Curiousity: why avoid it?

Reply:
LINQ is too expensive sometimes.